pyaedt/icepak.py

```diff
@@ -4190,39 +4136,30 @@ def assign_solid_block(
         ----------
         object_name : str or list
             Object name or a list of object names.
-        power_assignment : str or dict
+        power_assignment : str or dict or BoundaryDictionary
             String with the value and units of the power assignment or with
-            ``"Joule Heating"``. For a temperature-dependent or transient
-            assignment, a dictionary can be used. The dictionary should contain three keys:
-            ``"Type"``, ``"Function"``, and ``"Values"``.
-            - For the ``"Type"`` key, accepted values are ``"Temp Dep"`` and ``"Transient"``.
-            - For the ``"Function"`` key, acceptable values depend on the ``"Type"`` key
-            selection. When the ``"Type"`` key is set to ``"Temp Dep"``, the only
-            accepted value is ``"Piecewise Linear"``. When the ``"Type"`` key is
-            set to ``"Transient"``, acceptable values are `"Exponential"``, `"Linear"``,
-            ``"Piecewise Linear"``, ``"Power Law"``, ``"Sinusoidal"``, and ``"SquareWave"``.
-            - For the ``"Values"`` key, a list of strings contain the parameters required by
-            the ``"Function"`` key selection. For example, when``"Linear"`` is set as the
-            ``"Function"`` key, two parameters are required: the value of the variable
-            at t=0 and the slope of the line. For the parameters required by each
-            ``"Function"`` key selection, see the Icepak documentation. The parameters
-            must contain the units where needed.
+            ``"Joule Heating"``.
+            Assign a transient condition using the result of a function with
+            the pattern ``create_*_transient_assignment``.
+            Assign a temperature-dependent condition using the result of a
+            function with the pattern ``create_temp_dep_assignment``.
         boundary_name : str, optional
             Name of the source boundary. The default is ``None``, in which case the
             boundary name is automatically generated.
-        htc : float, str, or dict, optional
+        htc : float, str, or dict or BoundaryDictionary, optional
             String with the value and units of the heat transfer coefficient for the
-            external conditions. If a float is provided, the ``"w_per_m2kel"`` unit is used.
-            For a temperature-dependent or transient
-            assignment, a dictionary can be used. For more information, see the
-            description for the preceding ``power_assignment`` parameter. The
-            default is ``None``, in which case no external condition is applied.
-        ext_temperature : float, str or dict, optional
+            external conditions. If a float is provided, the unit is ``"w_per_m2kel"``.
+            Assign a transient condition using the result of a function with
+            the pattern ``create_*_transient_assignment``.
```

Comment:
```suggestion
            the ``create_*_transient_assignment`` pattern .
```
Would you please search and do a serach and replace for this edit? I'm not going to add suggestions for the other occurrences.